x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/ml_inference_logic.ts

```diff
@@ -385,6 +387,12 @@ export const MLInferenceLogic = kea<
     ],
   },
   selectors: ({ selectors }) => ({
+    filterModelId: [
```

Comment:
With my limited UI knowledge I'm not sure if it's a best practice to introduce a new selector for this (@TattdCodeMonkey let me know if you disagree). I'd rather just keep it simple and update the `existingInferencePipelines` selector's filter (line 540 in my local code):
```
.filter((p): p is MLInferencePipelineOption => p !== undefined && !isTextExpansionModel(p.modelType));
```

And declare `isTextExpansionModel()` as `!!model?.inference_config?.text_expansion`.

Reply:
Yep! I'll try this out. Thanks Adam !

Reply:
I agree with Adam we should not need a new selector for `filterModelId`, the `existingInferencePipelines` selector is building the list of pipelines we display, we should update that selector to filter our any pipelines that use a `text_expansion` model. We should already have all the data we need in that selector do remove those pipelines from the result.